lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -409,18 +429,29 @@ defmodule KernelTest do
 
       result = expand_to_string(quote(do: rand() in [1, 2]))
       assert result =~ "var = rand()"
-      assert result =~ ":erlang.or(:erlang.\"=:=\"(var, 2), :erlang.\"=:=\"(var, 1))"
+      assert result =~ ":erlang.orelse(:erlang.\"=:=\"(var, 2), :erlang.\"=:=\"(var, 1))"
 
       result = expand_to_string(quote(do: rand() in [1 | [2]]))
       assert result =~ "var = rand()"
-      assert result =~ ":erlang.or(:erlang.\"=:=\"(var, 1), :erlang.\"=:=\"(var, 2))"
+      assert result =~ ":erlang.orelse(:erlang.\"=:=\"(var, 1), :erlang.\"=:=\"(var, 2))"
+
+      result = expand_to_string(quote(do: rand() in [1 | some_call()]))
+      assert result =~ "var = rand()"
+      assert result =~ "{var0} = {some_call()}"
+      assert result =~ ":erlang.orelse(:erlang.\"=:=\"(var, 1), :lists.member(var, var0))"
     end
 
     defp expand_to_string(ast) do
       ast
       |> Macro.prewalk(&Macro.expand(&1, __ENV__))
       |> Macro.to_string
     end
+
+    defp assert_eval_string_error(error, msg, string) do
```

Comment:
I think using `exception` instead of `error` would be more descriptive.
We use that in `assert_receive` as well.

Perhaps `msg` –> `message` for better consistency?